### PR TITLE
SBTWO-9053 Updated middleware to handle request header objects

### PR DESCRIPTION
**Pull request by WarwickGoatee**
Not all req.headerses are functions apparently....

### CODE REVIEW

**Comment by lol768** [comment]
What about the XRU header reads?

https://github.com/UniversityofWarwick/warwick-trusted-apps/blob/f3c10ca3e5d6f9b9bc00a90c1945ef122fd72903/lib/trusted-apps.js#L20-L26

**Review by lol768 — CHANGES_REQUESTED** [review]
Might make sense to put in a helper function given the XRU checks too?

File: lib/trusted-apps.js [hunk 1]
```diff
@@ -43,10 +43,19 @@ function middleware(req, res, next) {
...
+  if (typeof req.header === 'function') {
+    certificate = req.header(HEADER_CERTIFICATE);
+    providerId = req.header(HEADER_PROVIDER_ID);
+    signature = req.header(HEADER_SIGNATURE);
+  } else if (req.header && typeof req.header === 'object') {
+    certificate = req.header[HEADER_CERTIFICATE];
+    providerId = req.header[HEADER_PROVIDER_ID];
+    signature = req.header[HEADER_SIGNATURE];
+  } else {
+    return error(res, 'header-invalid-type', 'req.header is not a function or object');
```
Comment (cursley) [inline lib/trusted-apps.js]: We could throw an exception here – this doesn't feel like a user-facing Unauthorized error so much as a development-time framework incompatibility

Comment (WarwickGoatee) [reply lib/trusted-apps.js]: Done it :)

File: lib/trusted-apps.js [hunk 2]
```diff
@@ -18,10 +18,20 @@ function setConfig(config) {
...
+    }
+  } else if (req.header && typeof req.header === 'object') {
+    if (req.header['X-Requested-URI']) {
+      return req.header['X-Requested-URI'];
+    } else {
+      return req.protocol + '://' + req.header['Host'] + req.url;
+    }
   } else {
-    return req.protocol + '://' + req.header('Host') + req.url;
+    throw "req.header is not a function or an object";
```
Comment (lol768) [inline lib/trusted-apps.js]:
```suggestion
    throw new Error("req.header is not a function or an object");
```

**Review by matmannion — APPROVED** [review]

**Review by lol768 — APPROVED** [review]

**Comment by WarwickGoatee** [comment]
Just doing the helper function thing...

**Comment by WarwickGoatee** [comment]
There

**Pull request merged**